tests/utils.py

```diff
@@ -135,6 +136,26 @@ def running_on_jenkins() -> bool:
     return RUNNING_ON_JENKINS
 
 
+def multithreaded_run(num_threads: int = 5) -> None:
+    """When multithreading_mode is enabled, run the decorated test function in multiple threads."""
+    from tests.conftest import MULTITHREADING_TEST_MODE_ENABLED
+
+    def decorator(func):
+        @pytest.mark.multithreaded_run
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+            if MULTITHREADING_TEST_MODE_ENABLED:
```

Comment:
for those tests, probably doesn't work when the thread safe session object is not enabled, you might have to add a check here once you have the feature flag

Reply:
I seem to have removed config which creates session in thread-safe mode with `--multithreading_mode` option. I have readded it now.